src/cli/add.rs

```diff
@@ -217,9 +217,23 @@ pub async fn add_pypi_specs_to_project(
         // TODO: Get best version
         // Add the dependency to the project
         if specs_platforms.is_empty() {
+            let dependencies = project.pypi_dependencies(None);
+            if let Some(package_name) = dependencies.keys().find(|pkg_name| &name == pkg_name) {
+                return Err(miette::miette!(
+                    "{} is already added as dependency.",
+                    console::style(package_name.as_source_str()).bold(),
+                ));
+            }
             project.manifest.add_pypi_dependency(name, spec, None)?;
         } else {
             for platform in specs_platforms.iter() {
+                let dependencies = project.pypi_dependencies(Some(*platform));
+                if let Some(package_name) = dependencies.keys().find(|pkg_name| &name == pkg_name) {
+                    return Err(miette::miette!(
+                        "{} is already added as dependency.",
+                        console::style(package_name.as_source_str()).bold(),
+                    ));
+                }
```

Comment:
I guess you could add these checks to `add_(pypi_)dependency`

Reply:
I can't access the `dependencies` from that context (`Manifest`). Is there a way of doing that? I'm guessing passing `dependencies` as argument is not optimal.

Reply:
nvm I think I found a way